src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -934,40 +1025,71 @@ private void clearSourceFromLayer(int destId) throws Exception {
 	 *
 	 * @param getSourceIndex new source index
 	 * @param currentScreenDestId name of current super/screen destination
-	 * @param layerId layer id
+	 * @param layerIdx index of layer
 	 * @param isMixedLayer check if layer is mixed
 	 * @throws Exception Throw exception when fail to changeContent
 	 */
-	private void assignToDest(int getSourceIndex, int currentScreenDestId, int layerId, boolean isMixedLayer) throws Exception {
+	private void assignToDest(int getSourceIndex, int currentScreenDestId, int layerIdx, boolean isMixedLayer) throws Exception {
 		// STEP 1 get dimension from source and destination
 		Dimension sourceDimension = populateDimension(getSourceIndex, true);
 		Dimension destDimension = populateDimension(currentScreenDestId, false);
 		// STEP 2 calculate dimension to make the layer stay in the middle of the screen destination
 		Dimension layerDimension = calculateProperDimension(sourceDimension, destDimension);
 		// STEP 3 put dimension to list
 		List<LayerDTO> layerDTOList = new ArrayList<>();
+		// GET DEST CONTENT
+		JsonNode currenScreenDestContent = getScreenDestContent(currentScreenDestId);
+		JsonNode layerNode = currenScreenDestContent.get(BarcoE2Constant.LAYERS);
 		if (isMixedLayer) {
-			LayerDTO firstLayerDTO = new LayerDTO(layerId, getSourceIndex, 1, 0);
-			firstLayerDTO.setDimension(layerDimension);
+			LayerDTO firstLayerDTO = prepareLayerDTO(layerIdx, getSourceIndex, 1, 0, layerDimension, layerNode);
 			layerDTOList.add(firstLayerDTO);
-			LayerDTO secondLayerDTO = new LayerDTO(layerId + 1, getSourceIndex, 0, 1);
-			secondLayerDTO.setDimension(layerDimension);
+			LayerDTO secondLayerDTO = prepareLayerDTO(layerIdx + 1, getSourceIndex, 0, 1, layerDimension, layerNode);
 			layerDTOList.add(secondLayerDTO);
 		} else {
-			LayerDTO layerDTO = new LayerDTO(layerId, getSourceIndex, 1, 1);
-			layerDTO.setDimension(layerDimension);
+			LayerDTO layerDTO = prepareLayerDTO(layerIdx, getSourceIndex, 1, 1, layerDimension, layerNode);
 			layerDTOList.add(layerDTO);
 		}
 		// STEP 4 prepare param before call changeContent
 		Map<Object, Object> changeScreenParams = new HashMap<>();
 		changeScreenParams.put(BarcoE2Constant.ID, currentScreenDestId);
-		changeScreenParams.put(BarcoE2Constant.LAYERS, layerDTOList.toArray());
-		JsonNode changeScreenContentResponse = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams);
-		if (changeScreenContentResponse.get(BarcoE2Constant.SUCCESS_STATUS) == null || changeScreenContentResponse.get(BarcoE2Constant.SUCCESS_STATUS).asInt() != 0) {
+		boolean result;
+		if (isMixedLayer) {
+			changeScreenParams.put(BarcoE2Constant.LAYERS, List.of(layerDTOList.get(0)).toArray());
+			boolean assignFirstLayer = requestByMethod(BarcoE2Constant.METHOD_CHANGE_CONTENT, changeScreenParams).get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+			changeScreenParams.remove(BarcoE2Constant.LAYERS);
```

Comment:
won't this clear anything that's been added at line 1057? 
If line 1057 is there for merely referencing purposes (of passing to the requestByMethod() ) - maybe it would make more sense to pass List.of(layerDTOList.get(0)).toArray() there directly?

Reply:
since the RPC request requires 2 params (1 for id and 1 for layerDTO) in it, so I clear this one (line 1057) after the first execution.

Reply:
Ok, I see, so you are just reusing the rpc body, got it :)